crates/rpc/ipc/src/stream_codec.rs

```diff
@@ -115,7 +115,8 @@ impl tokio_util::codec::Decoder for StreamCodec {
                 if depth == 0 && idx != start_idx && idx - start_idx + 1 > whitespaces {
                     let bts = buf.split_to(idx + 1);
                     return match String::from_utf8(bts.into()) {
-                        Ok(val) => Ok(Some(val)),
+                        // Trim to remove surrounding whitespace from a valid JSON block
+                        Ok(val) => Ok(Some(val.trim().to_string())),
```

Comment:
this now does another allocation, I don't think this is worth doing because serde_json deserialze can handle this case I believe, but I would accept a test for this when this returns leading whitespace